File: XMLCheck/Handler/CheckRulesXML01/Check_MA_DOITUONG_KCB.cs
using System.Text.Json;
using XMLCheck.Handler.CheckRulesXML1;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML01
{
    public class Check_MA_DOITUONG_KCB : Xml1RuleBase
    {
        public override string Key => "MA_DOITUONG_KCB";

        public override ErrorDetails? Check(Xml1Model model, HoSoContext context)
        {
            // 1. Không được để trống
            if (string.IsNullOrWhiteSpace(model.MA_DOITUONG_KCB))
            {
                return Error("MA_DOITUONG_KCB không được để trống");
            }

            if (!DictionaryCache.DoiTuongKCB.ContainsKey(model.MA_DOITUONG_KCB))
            {
                return Error($"{Key} không tồn tại trong danh mục");
            }
            return null;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml1/Check_MA_DOITUONG_KCB.ts
import { Xml1RuleBase } from './xml1RuleBase';
import { Xml1Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { DictionaryCacheInstance } from '../../dictionary.cache';

export class Check_MA_DOITUONG_KCB extends Xml1RuleBase
    {
  constructor() {
    super({
      ruleId: 'RULE_XML1_MA_DOITUONG_KCB_01',
      xmlType: 'XML1',
      field: 'MA_DOITUONG_KCB',
      severity: 'ERROR',
      description: 'Kiểm tra quy định cho trường MA_DOITUONG_KCB',
      errorMessage: 'MA_DOITUONG_KCB không được để trống',
      reference: 'QĐ 3176/QĐ-BYT',
    });
  }

        

        public check(model: Xml1Model, context: HoSoContext): ErrorDetails | null
        {
            // 1. Không được để trống
            if (!model.MA_DOITUONG_KCB || !model.MA_DOITUONG_KCB.trim())
            {
                return this.error("MA_DOITUONG_KCB không được để trống");
            }

            if (!(model.MA_DOITUONG_KCB in DictionaryCacheInstance.doiTuongKCB))
            {
                return this.error(`${this.key} không tồn tại trong danh mục`);
            }
            return null;
        }
    }
